### PR TITLE
Scripting: Move script_cache into _nodes/stats

### PR DESCRIPTION
**Pull request by stu-elastic**
Pending: #59262

Updated `_nodes/stats`:
 * Remove `script_cache`
 * Update `script` in `_node/stats` to include stats per context:

```
      "script": {
        "compilations": 1,
        "cache_evictions": 0,
        "compilation_limit_triggered": 0,
        "contexts":[
          {
            "context": "aggregation_selector",
            "compilations": 0,
            "cache_evictions": 0,
            "compilation_limit_triggered": 0
          },

```

Refs: #50152

### CODE REVIEW

**Comment by stu-elastic** [comment]
@elastic-machine test elasticsearch-ci/1

**Comment by stu-elastic** [comment]
@elasticmachine test elasticsearch-ci/1

**Comment by stu-elastic** [comment]
`/gradlew ':server:test' --tests "org.elasticsearch.cluster.coordination.FollowersCheckerTests.testFailsNodeThatIsUnhealthy" -Dtests.seed=3D481AA42BA5E7F9 -Dtests.security.manager=true -Dtests.locale=sq -Dtests.timezone=America/Argentina/Tucuman -Druntime.java=11` 

fails in master

See: https://gradle-enterprise.elastic.co/s/inzdodcrold2s

**Comment by stu-elastic** [comment]
@elasticmachine test elasticsearch-ci/1

File: server/src/test/java/org/elasticsearch/action/admin/cluster/node/stats/NodeStatsTests.java [hunk 1]
```diff
@@ -241,11 +243,35 @@ public void testSerialization() throws IOException {
                     }
                 }
                 ScriptStats scriptStats = nodeStats.getScriptStats();
+                ScriptStats deserializedScriptCacheStats = deserializedNodeStats.getScriptStats();
```
Comment (stu-elastic) [inline server/src/test/java/org/elasticsearch/action/admin/cluster/node/stats/NodeStatsTests.java:246]: `deserializedScriptCacheStats` -> `deserializedScriptStats`

**Review by rjernst — APPROVED** [review]
LGTM

**Review by jdconrad — APPROVED** [review]

File: server/src/test/java/org/elasticsearch/action/admin/cluster/node/stats/NodeStatsTests.java [hunk 2]
```diff
@@ -514,20 +525,10 @@ public static NodeStats createNodeStats() {
...
-                String context = randomValueOtherThanMany(stats::containsKey, () -> randomAlphaOfLength(12));
-                stats.put(context, new ScriptStats(randomLongBetween(0, 1024), randomLongBetween(0, 1024), randomLongBetween(0, 1024)));
-            }
-            scriptCacheStats = new ScriptCacheStats(stats);
-        }
         //TODO NodeIndicesStats are not tested here, way too complicated to create, also they need to be migrated to Writeable yet
         return new NodeStats(node, randomNonNegativeLong(), null, osStats, processStats, jvmStats, threadPoolStats,
-                fsInfo, transportStats, httpStats, allCircuitBreakerStats, scriptStats, discoveryStats,
-                ingestStats, adaptiveSelectionStats, scriptCacheStats);
+            fsInfo, transportStats, httpStats, allCircuitBreakerStats, scriptStats, discoveryStats,
```
Comment (stu-elastic) [inline server/src/test/java/org/elasticsearch/action/admin/cluster/node/stats/NodeStatsTests.java:530]: Add indent back.

**Pull request merged**